motioneye/handlers/config.py

```diff
@@ -394,6 +394,19 @@ def _handle_get_config_response(
         else:
             resp.remote_ui_config['id'] = camera_id
 
+            # sync admin_only flag locally to match the remote state
+            local_config.setdefault('@admin_only', False)
+            admin_only = bool(resp.remote_ui_config.get('admin_only', False))
+
+            if local_config.get('@admin_only') != admin_only:
+                local_config['@admin_only'] = admin_only
+                config.set_camera(camera_id, local_config)
```

Comment:
Since `config.set_camera(camera_id, local_config)` could be called twice now, while handling a single response, it makes sense to just set a flag here and in the condition below, and apply the config only once, before return, if needed.